xt/testapp/lib/Saml2Test.pm

```diff
@@ -164,39 +176,104 @@ get '/logout-soap' => sub {
 
     my $res = $soap->request($logoutreq);
 
-    redirect '/', 302;
+    if ($res) {
+        my $logout = Net::SAML2::Protocol::LogoutResponse->new_from_xml(
+            xml => $res
+        );
+        if ($logout->success) {
+            print STDERR "\nLogout Success Status - $logout->{issuer}\n";
+        }
+    }
+    else {
+        return "<html><pre>Bad Logout Response</pre></html>";
+    }
+
+    redirect '/?logout=SOAP', 302;
     return "Redirected\n";
 };
 
 post '/consumer-post' => sub {
     my $post = Net::SAML2::Binding::POST->new(
         cacert => config->{cacert},
     );
+
     my $ret = $post->handle_response(
         params->{SAMLResponse}
     );
 
     if ($ret) {
+
         my $assertion = Net::SAML2::Protocol::Assertion->new_from_xml(
             xml         => decode_base64(params->{SAMLResponse}),
             key_file    => config->{key},
             cacert      => config->{cacert},
         );
 
+        if ($assertion->{response_status} ne 'urn:oasis:names:tc:SAML:2.0:status:Success') {
```

Comment:
Idea: Make $assertion->success() so we have this everywhere?

Reply:
should be assertion->valid I will take a look

Reply:
Fixed with $assertion->valid